geoh5py/shared/entity_type.py

```diff
@@ -132,3 +132,9 @@ def workspace(self) -> ws.Workspace:
         # which means workspace has been deleted.
         assert workspace is not None
         return workspace
+
+    @workspace.setter
```

Comment:
We don't want a setter on the workspace as it should not be changed by the user. The responsibility is on the workspace to manage the unique entity_types

Reply:
But as workspace is weakref, once it's closed, it's None.
We need to copy the entity to the new workspace! 
at least create a copy function for entity_type?

Reply:
Yep that would be nicer. I would also need a check in the data creation to make sure that the workspace of entity_type is the same as parent object, else - call the copy method.

Reply:
Created a copy function in entity_type